domain/externalcontroller/state/state.go

```diff
@@ -32,116 +33,86 @@ func (st *State) Controller(
 	ctx context.Context,
 	controllerUUID string,
 ) (*crossmodel.ControllerInfo, error) {
-
 	db, err := st.DB()
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 
-	var (
-		controllerInfo crossmodel.ControllerInfo
-		controllerRow  *sql.Rows
-	)
-	err = db.StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
-		q := `
-SELECT 	alias, ca_cert, address  
-FROM 	external_controller AS ctrl
-	LEFT JOIN external_controller_address AS addrs
-	ON ctrl.uuid = addrs.controller_uuid
-WHERE 	ctrl.uuid=?`[1:]
-
-		controllerRow, err = tx.QueryContext(ctx, q, controllerUUID)
-		if err != nil {
-			return errors.Trace(err)
-		}
-		controllerInfo, err = controllerInfoFromRow(controllerUUID, controllerRow)
-		return err
-	})
+	s := sqlair.MustPrepare(`
+SELECT (alias, ca_cert, address) as &ExternalController.* 
+FROM   external_controller AS ctrl
+       LEFT JOIN external_controller_address AS addrs
+       ON ctrl.uuid = addrs.controller_uuid
+WHERE  ctrl.uuid=$M.id`, ExternalController{}, sqlair.M{})
```

Comment:
is it possible to have the pointer in the argument instead of in the sql query?:
```go
	s := sqlair.MustPrepare(`
SELECT (alias, ca_cert, address) as ExternalController.* 
FROM   external_controller AS ctrl
       LEFT JOIN external_controller_address AS addrs
       ON ctrl.uuid = addrs.controller_uuid
WHERE  ctrl.uuid=$M.id`, &ExternalController{}, sqlair.M{})
```
It makes the SQL statements a bit more readable

Reply:
No. That is the SQLair syntax signalling with "&" that it is an output argument, and with "$" that it is an input.